sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -247,7 +249,53 @@ internal DataFeedDetailPatch GetPatchModel()
             patch.Admins = AdministratorsEmails;
             patch.Viewers = ViewersEmails;
 
+            SetAuthenticationProperties(patch, DataSource);
+
             return patch;
         }
+
+        private static void SetAuthenticationProperties(DataFeedDetail detail, DataFeedSource dataSource)
+        {
+            switch (dataSource)
+            {
+                case AzureBlobDataFeedSource s:
+                    detail.AuthenticationType = s.GetAuthenticationTypeEnum();
+                    break;
+                case AzureDataExplorerDataFeedSource s:
+                    detail.AuthenticationType = s.GetAuthenticationTypeEnum();
+                    detail.CredentialId = s.DatasourceCredentialId;
+                    break;
+                case AzureDataLakeStorageGen2DataFeedSource s:
+                    detail.AuthenticationType = s.GetAuthenticationTypeEnum();
+                    detail.CredentialId = s.DatasourceCredentialId;
+                    break;
+                case SqlServerDataFeedSource s:
+                    detail.AuthenticationType = s.GetAuthenticationTypeEnum();
+                    detail.CredentialId = s.DatasourceCredentialId;
+                    break;
```

Comment:
These `AuthenticationType` are all different types (`AzureBlobDataFeedSource.AuthenticationType`, `AzureDataExplorerDataFeedSource.AuthenticationType`, etc), while the service uses a single type: `GetAuthenticationTypeEnum` (internal).

We decided to break this enum into multiple enums in .NET to make it easier for users to know which authentication types are valid for each data source. For instance, `AzureBlobDataFeedSource.AuthenticationType` only has `Basic` and `ManagedIdentity`. Users will never make the mistake of trying to set a `ServicePrincipal` authentication in this data source.

According to Ted, auth is one of most confusing things for users starting to get used to the API (as a user myself, I can totally agree), so we decided to split it like this to make it easier for them.